### PR TITLE
Feature/add upper limit on submit quotes

### PR DESCRIPTION
**Pull request by wjuan-mob**
Limit the number of quotes that can be simultaneously submitted.

### CODE REVIEW

File: server/src/client_service.rs [hunk 1]
```diff
@@ -455,6 +470,88 @@ mod tests {
...
+        let pair = Pair {
+            base_token_id: TokenId::from(1),
+            counter_token_id: TokenId::from(2),
+        };
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let quote_book = InMemoryQuoteBook::default();
+        let (client_api, _server, _msg_bus_tx, _msg_bus_rx) =
+            create_test_client_and_server(&quote_book, &logger);
+
+        // Submitting SCIs under MAX_Simultaneous_Quote should be successful
```
Comment (eranrund) [inline server/src/client_service.rs:484]:
```suggestion
        // Submitting SCIs under MAX_SIMULTANEOUS_QUOTES should be successful
```

File: server/src/client_service.rs [hunk 2]
```diff
@@ -242,7 +245,19 @@ impl<OB: QuoteBook> DeqsClientApi for ClientService<OB> {
     ) {
         let _timer = SVC_COUNTERS.req(&ctx);
         scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
-            send_result(ctx, sink, self.submit_quotes_impl(req, logger), logger)
+            let result: Result<SubmitQuotesResponse, RpcStatus> =
+                if req.quotes.len() >= MAX_SIMULTANEOUS_QUOTES {
```
[the remark this answers is not in the document]
Comment (wjuan-mob) [reply server/src/client_service.rs:249]: No, I will move it. My thought process was just that it seemed like a different sort of logical check than what was happening inside submit_quotes_impl so I just put it directly into submit quotes.

**Comment by wjuan-mob** [comment]
Current dependencies on/for this PR:
* main
  * **PR #44** <a href="https://app.graphite.dev/github/pr/mobilecoinofficial/deqs/44" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinofficial/deqs/44?utm_source=stack-comment).

File: server/src/client_service.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (wjuan-mob) [reply server/src/client_service.rs]: Thank you for explaining. That makes sense to me.

**Review by eranrund — CHANGES_REQUESTED** [review]

File: server/src/client_service.rs (hunk 1 above)
Comment (wjuan-mob) [reply server/src/client_service.rs]: Yeah, I'm not sure why that happened. I think maybe I force pushed over it? Sorry about that.

**Review by eranrund — APPROVED** [review]

**Pull request merged**